processing/src/main/java/io/druid/segment/incremental/IncrementalIndexStorageAdapter.java

```diff
@@ -701,7 +716,20 @@ public boolean matches()
     }
 
     @Override
-    public ValueMatcher makeValueMatcher(String dimension, final Predicate predicate)
+    public ValueMatcher makeValueMatcher(String dimension, final DruidPredicateFactory predicateFactory)
+    {
+      ValueType type = getTypeForDimension(dimension);
+      switch (type) {
+        case LONG:
+          return makeLongValueMatcher(dimension, predicateFactory.makeLongPredicate());
+        case STRING:
+          return makeStringValueMatcher(dimension, predicateFactory.makeStringPredicate());
+        default:
+          throw new UOE("Cannot make ValueMatcher for type[%s]", type);
```

Comment:
Similar comment to QueryableIndexStorageAdapter, this should probably act like the column is a string column full of nulls when we cannot filter on that particular column type.